terraform/modules/neptune/variables.tf

```diff
@@ -0,0 +1,116 @@
+variable "resource_prefix" {
+  description = "the prefix to add when creating resources"
+  type        = string
+}
+
+variable "tags" {
+  description = "tags to associate with this instance"
+  type = map(string)
+}
+
+variable "deletion_protection" {
+  description = "prevent deletion"
+  type        = bool
+  default     = false
+}
+
+variable "backup_retention_period" {
+  description = "number of days to keep backup"
+  type        = number
+  default     = 35
+}
+
+variable "neptune_engine_version" {
+  description = "aurora database engine version."
+  type        = string
+  default     = "1.2.0.1"
+}
+
+variable "neptune_iam_roles" {
+  description = "A List of ARNs for the IAM roles to associate to the Neptune Cluster."
+  type        = list(string)
+  default     = []
+}
+
+variable "create_neptune_subnet_group" {
+  type        = bool
+  description = "whether to create a neptune subnet group for the rds instance"
+  default     = true
+  sensitive   = false
+}
+
+variable "db_subnet_group_name" {
+  type        = string
+  description = "name of the db subnet group - required if create_db_subnet_group is false"
+  default     = null
+  sensitive   = false
+}
+
+variable "backup_window" {
```

Comment:
We should talk to Mike about coming up with standards for things like maintenance windows and back-up windows across the program and across DB types (open search, rds, etc.).